lesson24/index.js

```diff
@@ -0,0 +1,46 @@
+const checkboxLink = document.getElementById("js-checkbox-link");
+const modalCloseButton = document.getElementById("js-modal-close-button");
+const modalInner = document.getElementById("js-modal-inner");
+const submitButton = document.getElementById("js-submit-button");
+
+const openModal = () => {
+    document.getElementById("js-modal-area").classList.add("is-active");
+};
+
+const closeModal = () => {
+    document.getElementById("js-modal-area").classList.remove("is-active");
+};
+
+checkboxLink.addEventListener("click", openModal);
+modalCloseButton.addEventListener("click", closeModal);
+
+//モーダル以外の部分を押すとモーダルが閉じる
+document.addEventListener("click", (e) => {
+    e.target.classList.contains("js-modal") && closeModal();
+});
+
+//スクロールが一番下に行ったらチェックボックスをcheckedにする
+const observerTarget = document.getElementById("js-last-sentence");
+const observerOptions = {
+    root: modalInner,
+    rootMargin: "0px",
+    threshold: 1.0
+};
+const checkbox = document.getElementById("js-checkbox");
+
+const setCheckedAttributeToCheckbox = entries => {
+    entries.forEach(entry => {
```

Comment:
IMO
If there is only one target element, it is possible to use [Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to retrieve only the first element.

```suggestion
const setCheckedAttributeToCheckbox = ([ entry ]) => {
```

Reply:
Thank you for telling me!
I am not good at Destructuring assignments, so I had to relearn them.
I understood that when `[ entry ]` is passed as an argument, it is processed as shown below. Am I right?
```
const setCheckedAttributeToCheckbox = entries => {
    const [entry] = entries; //here
    if (entry.isIntersecting) {
        checkbox.checked = true;
        checkbox.disabled = false;
    }
};
```
[d001cb9](https://github.com/haru-programming/JavaScript-lessons/pull/28/commits/d001cb99c86356cc9f219f1df7e60f85e9a4ad27)[d001cb9](https://github.com/haru-programming/JavaScript-lessons/pull/28/commits/d001cb99c86356cc9f219f1df7e60f85e9a4ad27)

Reply:
Thank you for telling me!
I am not good at Destructuring assignments, so I had to relearn them.
I understood that when `[ entry ]` is passed as an argument, it is processed as shown below. Am I right?
```
const setCheckedAttributeToCheckbox = entries => {
    const [entry] = entries; //here
    if (entry.isIntersecting) {
        checkbox.checked = true;
        checkbox.disabled = false;
    }
};
```
[d001cb9](https://github.com/haru-programming/JavaScript-lessons/pull/28/commits/d001cb99c86356cc9f219f1df7e60f85e9a4ad27)[d001cb9](https://github.com/haru-programming/JavaScript-lessons/pull/28/commits/d001cb99c86356cc9f219f1df7e60f85e9a4ad27)

Reply:
You're right 👏